### PR TITLE
cluster: report running phase

### PR DESCRIPTION
**Pull request by xiang90**

### CODE REVIEW

File: pkg/cluster/cluster.go [hunk 1]
```diff
@@ -176,8 +176,21 @@ func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
 		}
 		close(c.stopCh)
 		wg.Done()
+
+		c.status.SetPhaseFailed()
+		// best effort to mark the cluster as failed
+		for {
```
Comment (hongchaodeng) [inline pkg/cluster/cluster.go]: Use retryutil.Retry()?

Comment (xiang90) [inline pkg/cluster/cluster.go]: sure.

**Comment by hongchaodeng** [comment]
@etcd-bot retest this pleas

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**